packages/fetchai/skills/tac_control_contract/game.py

```diff
@@ -87,8 +87,9 @@ def __init__(
         self._version_id = version_id
         self._tx_fee = tx_fee
         self._agent_addr_to_name: Dict[str, str] = defaultdict()
+        # The next  line creates a problem when we are trying to generate ids due to the fact that we haven't initialise the contract.
         self._good_id_to_name = generate_good_id_to_name(nb_goods, contract)
```

Comment:
The actual problem is that we are passing the contract as parameter in the `generate_good_id_to_name` function and the contract is not initialised yet.